man/ostree.repo-config.xml

```diff
@@ -93,6 +93,14 @@ Boston, MA 02111-1307, USA.
         to <literal>false</literal>.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>change-update-summary</varname></term>
+        <listitem><para>Boolean value controlling whether or not to
+        automatically update the summary file after any ref is added,
+        removed, or updated. Defaults to <literal>false</literal>.
```

Comment:
It's worth mentioning/comparing with `commit-update-summary` from 9e6ac6d822f4dded703a99912f629ea1e9b3bcbc in <https://github.com/ostreedev/ostree/pull/83>.  This is an extension of that.

I understand the argument for doing this; but do also keep in mind my original arguments for not doing it by default - systems may want to ensure that multiple refs are updated atomically.  I suppose though that is (mostly) supported here if one does the ref updates in a single ostree txn via `transaction_set_ref()`.

I am not sure if there was an explicit reason the `commit-update-summary` option was implemented in the `ostree commit` CLI - it seems to me it should be in the library in `ostree_repo_commit_transaction()` right?

Reply:
>It's worth mentioning/comparing with commit-update-summary from 9e6ac6d in #83. This is an extension of that.

Sure, I added another sentence.

>I understand the argument for doing this; but do also keep in mind my original arguments for not doing it by default - systems may want to ensure that multiple refs are updated atomically. I suppose though that is (mostly) supported here if one does the ref updates in a single ostree txn via transaction_set_ref().

Even with transactions the summary update won't be atomic. It will be updated for each ref update in `ostree_repo_commit_transaction()`. I suppose the alternative would be to teach eos-updater and flatpak to update the summary after an operation but I'm not sure it's worth the complexity that would entail. You'd probably want it to be a configurable option, but why would clients of ostree implement its own option? And perhaps the atomicity won't matter much in practice, since flatpak pulls most dependencies before the app ref, and at least for the USB case GNOME Software shouldn't show an app as able to be put on a USB until it's been fully installed. In the LAN case I think GNOME Software would show the app despite missing dependencies so you could run into errors if you try to install at exactly the wrong time.

>I am not sure if there was an explicit reason the commit-update-summary option was implemented in the ostree commit CLI - it seems to me it should be in the library in ostree_repo_commit_transaction() right?

I had the same question but didn't feel qualified to make that change.

Reply:
> it seems to me it should be in the library in ostree_repo_commit_transaction() right?

That struck me as odd too. We should open an issue for that.

Reply:
#1689